lib/cover.nix

```diff
@@ -0,0 +1,58 @@
+{ stdenv, lib, haskellLib }:
+{ ghc, name, src, testDerivations, toCoverDerivations, excludedModules ? [] }:
+
+let
+  buildWithCoverage = builtins.map (d: d.covered);
+
+  tests = buildWithCoverage testDerivations;
+  covering = buildWithCoverage toCoverDerivations;
```

Comment:
I guess there's a bit of a policy decision here: do we require people to explicitly enable coverage in `cabal.project`, or do we encourage using this thing, which essentially just forces turning it on with Nix. 

It would be simpler if we required the user to set `coverage: true`, then we could just use the `checks` derivations as they are.

Either way, worth a note explaining why we're doing it one way and not the other.